### PR TITLE
Fix table_view_base doxygen format

### LINKED ISSUE #11295: [DOC] ColumnView not clickable link in doxygen
## Report incorrect documentation

**Location of incorrect documentation**
https://docs.rapids.ai/api/libcudf/stable/classcudf_1_1detail_1_1table__view__base.html#details

**Describe the problems or issues found in the documentation**
The return type of the `column` method `ColumnView` should be a link to the definition of the `column_view` class.

I believe the issue is that in `table_view_base` `ColumnView` is in the template type. 
While in `table_view` its a type alias and most of the methods are inherited from the base.

**Steps taken to verify documentation is incorrect**
Not sure if there is a way in Doxygen to fix this.

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Fixes the formatting of the `cudf::table_view_base` class doxygen description.
https://docs.rapids.ai/api/libcudf/stable/classcudf_1_1detail_1_1table__view__base.html#details

Closes #11295 

New format is as follows:
table_view_base-doxygen

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/include/cudf/table/table_view.hpp [hunk 1]
```diff
@@ -38,13 +38,14 @@ namespace detail {
  * @brief Base class for a table of `ColumnView`s
  *
  * This class should not be used directly. Instead:
- * -`table_view` should be used for a table of `column_view`s
- * -`mutable_table_view` should be used for a table of `mutable_column_view`s
+ * - `table_view` should be used for a table of `column_view` types
```
Comment (ttnghia) [inline cpp/include/cudf/table/table_view.hpp:41]: I thought that `column_view` is just one "type", not multiple "types", isn't it? Similar for `mutable_column_view`.

Comment (davidwendt) [reply cpp/include/cudf/table/table_view.hpp:41]: Sorry, I don't understand your question. 
The template variable `ColumnView` can be either `column_view` or `mutable_column_view`

Comment (ttnghia) [reply cpp/include/cudf/table/table_view.hpp:41]: Oh I meant this:
```suggestion
 * - `table_view` should be used for a table of `column_view` type
 * - `mutable_table_view` should be used for a table of `mutable_column_view` type
```

**Review by ttnghia — APPROVED** [review]

File: cpp/include/cudf/table/table_view.hpp (hunk 1 above)
Comment (vyasr) [reply cpp/include/cudf/table/table_view.hpp]: I think the David's intent with the plural was to reflect that the table contains multiple `column_view`s, so perhaps that's the better plural form?

Comment (davidwendt) [reply cpp/include/cudf/table/table_view.hpp]: I just don't care for how that looks is all.

Comment (vyasr) [reply cpp/include/cudf/table/table_view.hpp]: I agree :slightly_smiling_face: Do whichever you like here, I was just trying to clear up any confusion. If we really want to be precise while still avoiding the ugliness, we could do something like "a table of `column_view` objects" instead.

Comment (davidwendt) [reply cpp/include/cudf/table/table_view.hpp]: I updated the screen shot with my latest attempt.

Comment (vyasr) [reply cpp/include/cudf/table/table_view.hpp]: I like that version.

File: cpp/include/cudf/table/table_view.hpp [hunk 2]
```diff
@@ -38,13 +38,14 @@ namespace detail {
...
- * -`mutable_table_view` should be used for a table of `mutable_column_view`s
+ * - `table_view` should be used for a table of columns of type `column_view`
+ * - `mutable_table_view` should be used for a table of columns of type `mutable_column_view`
  *
  * All public constructors and member functions of `table_view_base` are
  * available in both `table_view` and `mutable_table_view`.
  *
- * @tparam ColumnView The type of column view the table contains
+ * @tparam ColumnView The type of column view the table contains:
+ *                    expects column_view or mutable_column_view
```
Comment (vyasr) [inline cpp/include/cudf/table/table_view.hpp:48]:
```suggestion
 * @tparam ColumnView The type of column view the table contains.
 *                    Expects column_view or mutable_column_view
```

**Review by vyasr — APPROVED** [review]

**Review by karthikeyann — APPROVED** [review]
LGTM 👍

**Comment by ttnghia** [comment]
Rerun tests.

**Comment by ttnghia** [comment]
Rerun tests.

**Comment by davidwendt** [comment]
@gpucibot merge

**Pull request merged**